workflow/common/convert.go

```diff
@@ -58,23 +57,34 @@ func NewWorkflowFromWorkflowTemplate(templateName string, workflowMetadata *meta
 	return wf
 }
 
-func toWorkflow(typeMeta metav1.TypeMeta, objectMeta metav1.ObjectMeta, spec wfv1.WorkflowSpec) *wfv1.Workflow {
+func toWorkflow(cronWf wfv1.CronWorkflow, objectMeta metav1.ObjectMeta) *wfv1.Workflow {
 	wf := &wfv1.Workflow{
 		TypeMeta: metav1.TypeMeta{
 			Kind:       workflow.WorkflowKind,
-			APIVersion: typeMeta.APIVersion,
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: objectMeta.GetName() + "-",
-			Labels:       make(map[string]string),
-			Annotations:  make(map[string]string),
+			APIVersion: cronWf.TypeMeta.APIVersion,
 		},
-		Spec: spec,
+		ObjectMeta: objectMeta,
+		Spec:       cronWf.Spec.WorkflowSpec,
 	}
 
-	if instanceId, ok := objectMeta.GetLabels()[LabelKeyControllerInstanceID]; ok {
+	if instanceId, ok := cronWf.ObjectMeta.GetLabels()[LabelKeyControllerInstanceID]; ok {
 		wf.ObjectMeta.GetLabels()[LabelKeyControllerInstanceID] = instanceId
 	}
 
+	wf.Labels[LabelKeyCronWorkflow] = cronWf.Name
+	if cronWf.Spec.WorkflowMetadata != nil {
```

Comment:
is this covered by tests? looks new

Reply:
This is already covered by tests. It was simply moved from the caller function (which is covered by tests) to the helper function. The helper function is only used by two parent functions and they are both covered by tests now